### PR TITLE
Fix odoc seps

### PR DESCRIPTION
**Pull request by gpetiot**
Follows up on #509 

Fix unnecessary spaces that changes the formatting.

It is still necessary to modify the docstrings in the `infer` codebase (cc @jberdine), for example:
- sometimes `[A]x` is used and we are supposed to have a space in-between, just use a `-` if the space is not wanted
- sometimes there are unexpected `@`-tags that have to be escaped or put in code blocks
- sometimes `{A}` that are parsed as odoc styles where A is a custom style name and an argument is expected so there is a space between A and the (empty) argument

### CODE REVIEW

**Comment by jberdine** [comment]
I am not sure of exactly why the checks are tripping on the infer test, but there are at least some `\[` escape sequences within `[` `]` code blocks where the `\` is lost in the output. AFAIU, this is incorrect as the escape is necessary to produce a `[` character in a code block.

**Comment by gpetiot** [comment]
Seems to also fix #482

File: src/Fmt_odoc.ml [hunk 1]
```diff
@@ -12,6 +12,12 @@
 open Octavius.Types
 open Fmt
 
+let str s =
+  String.substr_replace_all s ~pattern:"@" ~with_:"\\@"
+  |> String.substr_replace_all ~pattern:"{" ~with_:"\\{"
+  |> String.substr_replace_all ~pattern:"}" ~with_:"\\}"
```
Comment (gpetiot) [inline src/Fmt_odoc.ml]: Maybe it makes sense to also add `[` and `]`

Comment (jberdine) [reply src/Fmt_odoc.ml]: It could very well. Might address one of my comments on your infer PR?

Comment (jberdine) [reply src/Fmt_odoc.ml]: Also, it might make sense to use `Base.String.Escaping` for this to avoid a potential performance problem.

**Comment by gpetiot** [comment]
I disabled the parsing of docstring by default.

There are still some issues for the escaped characters, for example `\g` is not modified but `\{` is changed into `{` that makes it a nightmare to format latex strings

File: src/Fmt_odoc.ml [hunk 2]
```diff
@@ -90,11 +89,11 @@ and fmt_text_elt = function
       hvbox 0 (wrap "{!modules:" "}" (list l "@," str))
   | Special_ref SRK_index_list -> str "{!indexlist}"
   | Target (s, l) ->
+      let target = Option.value_map s ~default:"" ~f:(fun s -> s ^ ":") in
+      let latex = String.is_empty target || String.equal target "latex:" in
```
Comment (jberdine) [inline src/Fmt_odoc.ml]: Is it clear that latex is special here, or should all `Target`s be left unescaped?

Comment (gpetiot) [reply src/Fmt_odoc.ml]: I didn't find any info suggesting that latex is a special case, so I don't escape any target anymore

**Comment by gpetiot** [comment]
I know that you already accepted the changes a long time ago but since the code changed I would rather wait for a last approval before merging this one.

**Comment by gpetiot** [comment]
Last commit should fix #513

**Comment by gpetiot** [comment]
I don't understand how test_branch can still fail on master even though the parsing of docstrings is now disabled by default

**Comment by jberdine** [comment]
I debugged this a bit, it seems that there is a problem (I'm not sure where) that causes ocamlformat to misidentify the project root, leading to the tests failing. Running an ocamlformat command from the `test` directory, mimicing what `make test` does:
```
$ ../_build/dev/src/ocamlformat.exe --no-version-check -i code/infer/infer/src/checkers/cost.ml --print-config |& grep parse-docstrings=
parse-docstrings=true (file ../.ocamlformat:3)
```
Note that test/code/infer/.git does not seem to have stopped the config search from picking up .ocamlformat from higher in the file system. If I explicitly add `--root code/infer`, it works.

**Pull request merged**